explorer-ui/src/components/ProfileView/Preferences/Preferences.jsx

```diff
@@ -14,176 +14,257 @@ import Geocode from "react-geocode";
+    // Get latitude & longitude from address.
+    Geocode.fromAddress(region.current.value).then(
+      (response) => {
+        const { lat, lng } = response.results[0].geometry.location;
+        setPosition({ lat: lat, lng: lng });
+        console.log(lat, lng);
+      },
+      (error) => {
+        console.error(error);
       }
-    };
-    preferences();
-    console.log("preferences posted!")
-  };
+    );
+
+    console.log("current location is: ", currLocation);
+  }
 
+//   // Send the profile information to the database
+//   function handleSubmit(event) {
+//     console.log("submitted");
+//     event.preventDefault();
+//     const preferences = async () => {
+//       try {
+//         const res = await axios.post(
+//           `${config.API_BASE_URL}/profileInfo`,
+//           {
+//             preferenceInfo: preferenceInfo,
+//           },
+//           { maxContentLength: Infinity, maxBodyLength: Infinity }
+//         );
+//       } catch (err) {
+//         alert(err);
+//         console.log(err);
+//       }
+//     };
+//     preferences();
+//     console.log("preferences posted!");
+//   }
 
   // RETURN
   return (
     <div className="preferences">
-        <NavBar />
-      <form className="questionaire" onSubmit={(event) =>handleSubmit(event)}>
+      <NavBar />
+      <div className="questionaire">  
+      {/* onSubmit={(event) => handleSubmit(event)}> */}
         <div className="employment">
           <h3>What best describes the field you work in?</h3>
-          <select id="occupation" name="occupation" ref={employmentOption} onClick={(event) => handleOccupationsSelected(event, employmentOption)}>
-          {
-            employment_fields.map((occupation, idx) =>(
-                <option >{occupation}</option>
-            ))
-          }
+          <select
+            id="occupation"
+            name="occupation"
+            ref={employmentOption}
+            onClick={(event) =>
+              handleOccupationsSelected(event, employmentOption)
+            }
+          >
+            {employment_fields.map((occupation, idx) => (
+              <option key={idx}>{occupation}</option>
+            ))}
           </select>
         </div>
         <div className="interests">
           <h3>Select the following interests that apply to you</h3>
           <div className="grid">
-          {
-            interestsArr.map((interest, idx) =>(
-                <div >
-            <button className="card" id={interest} onClick={(event) => handleInterestSelected(event, interest.id)}>{interest.string}</button>
-        </div>
-            ))
-          }
+            {interestsArr.map((interest, idx) => (
+              <div key={idx}>
+                <button
+                  className="card"
+                  id={interest}
+                  onClick={(event) =>
+                    handleInterestSelected(event, interest.id)
+                  }
+                >
+                  {interest.string}
```

Comment:
is interests an object ? if so, think update the naming of key string to say value or name

Reply:
yes interests is a JSON object! fixed